parts/components/TMC2130.cpp

```diff
@@ -67,7 +67,10 @@ void TMC2130::_Draw(bool bIsSimple)
 			glVertex3f(350,0,0);
 			glVertex3f(350,10,0);
 			glVertex3f(0,10,0);
-			glColor3f(1,1,1);
+			if (m_regs.defs.GCONF.en_pwm_mode)
```

Comment:
This is NOT thread safe - we are in the GL thread at this time and this value is arbitrarily read/written by the main AVR thread at any time.  This will cause undefined behaviour. 

The value should be extracted into a std::atomic_bool and synchronzied at the same time the register is updated, just like m_bEnable or m_bDrawStall

Reply:
I see, forgot about this. Changing this

Reply:
Added another bool. Maybe at some point it would make sense to have a copy of the entire register just for rendering.

Which other functions besides _Draw run in separate threads?

Reply:
Threads are fun /s ;)

It is only the GL operations that are run in a different thread context (with the exception of some of the more "specialized" parts like the uart_pty that has threads to manage the IO from the host OS). It is entirely possible to "share" the register set between the two using mutexes - but this gets *very* problematic if you expect to have any sort of time accuracy. I had to deliberately remove any mutexes from the GL operations as part of the work to get Klipper stable. Native atomic types are also lighter weight than a mutex :)

Reply:
I wouldn't actually wrap that in a mutex. I'd really wholesale clone it into a second struct just for rendering when it's updated,  given the size.

Not worth it yet though..

Reply:
Agreed :)